### PR TITLE
Enhancement/signal search filters

### PR DESCRIPTION
**Pull request by kevgliss**

### CODE REVIEW

**Comment by wssheldon** [comment]
I think the chips on the filter creation dialog dont display the name correctly.

File: src/dispatch/static/dispatch/src/entity/EntityFilterCombobox.vue [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+            No entities matching "
+            <strong>{{ search }}</strong
+            >"
+          </v-list-item-title>
+        </v-list-item-content>
+      </v-list-item>
+    </template>
+    <template v-slot:selection="{ item, index }">
+      <v-chip close @click:close="entities.splice(index, 1)">
+        <span v-if="item.entity"> {{ item.name }} </span>{{ item.name }}
```
Comment (wssheldon) [inline src/dispatch/static/dispatch/src/entity/EntityFilterCombobox.vue:30]:
```suggestion
        <span v-if="item.entity"> {{ item.value }} </span>{{ item.value }}
```

Comment (wssheldon) [reply src/dispatch/static/dispatch/src/entity/EntityFilterCombobox.vue:30]: Probably want the `entity_type.name` / `entity.value` so you can see the type and value of the entity and select

**Comment by wssheldon** [comment]
The first time I create a filter, I am unable to save it if I choose a deduplicate filter. I tried longer field names.

Screenshot 2023-02-15 at 4 30 11 PM

You can save it as a snooze rule and then edit it. but the first time it will not let me.

Also, it seems the OK button doesn't work on the datetime picker for the suppression filter creation. I have to click out of the component to achieve the same affect.

Nothing helpful appears in JS console.

**Comment by wssheldon** [comment]
This looks good to me, only the couple UI things stood out.

**Comment by wssheldon** [comment]
I think there is an opportunity to improve the UX on the filter creation dialog, specifically the first page. I think maybe we just need a help description of what selecting an entity or entity type does. "Select all entity/ entity types that should get filtered if they appear in the signal"

File: src/dispatch/signal/service.py [hunk 2]
```diff
@@ -238,100 +209,52 @@ def create_instance(*, db_session, signal_instance_in: SignalInstanceCreate) ->
...
-
-    if rule_tag_ids == signal_tag_ids:
-        supressed = True
-        signal_instance.suppression_rule_id = suppression_rule.id
-
-    db_session.commit()
-    return supressed
+        query = db_session.query(SignalInstance).filter(
+            SignalInstance.signal_id == signal_instance.signal_id
+        )
```
Comment (wssheldon) [inline src/dispatch/signal/service.py:231]:
```suggestion
        query = (
            db_session.query(SignalInstance)
            .join(SignalInstance.entities)
            .join(EntityType)
            .filter(SignalInstance.signal_id == signal_instance.signal_id)
        )
```

Resolves:

```bash
ERROR:The query does not contain model `EntityType`.:/Users/wshel/Projects/dispatch/src/dispatch/signal/scheduled.py:consume_signals:48
Traceback (most recent call last):
  File "/Users/wshel/Projects/dispatch/src/dispatch/signal/scheduled.py", line 41, in consume_signals
    signal_flows.create_signal_instance(
  File "/Users/wshel/Projects/dispatch/src/dispatch/signal/flows.py", line 41, in create_signal_instance
    if signal_service.apply_filter_actions(db_session=db_session, signal_instance=signal_instance):
  File "/Users/wshel/Projects/dispatch/src/dispatch/signal/service.py", line 233, in apply_filter_actions
    query = apply_filters(query, f.expression, model_cls=SignalInstance)
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 319, in apply_filters
    sqlalchemy_filters = [filter.format_for_sqlalchemy(query, default_model) for filter in filters]
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 319, in <listcomp>
    sqlalchemy_filters = [filter.format_for_sqlalchemy(query, default_model) for filter in filters]
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 149, in format_for_sqlalchemy
    *[filter.format_for_sqlalchemy(query, default_model) for filter in self.filters]
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 149, in <listcomp>
    *[filter.format_for_sqlalchemy(query, default_model) for filter in self.filters]
  File "/Users/wshel/Projects/dispatch/src/dispatch/database/service.py", line 118, in format_for_sqlalchemy
    model = get_model_from_spec(filter_spec, query, default_model)
  File "/Users/wshel/.pyenv/versions/3.9.7/envs/dispatch/lib/python3.9/site-packages/sqlalchemy_filters/models.py", line 114, in get_model_from_spec
    raise BadSpec(
sqlalchemy_filters.exceptions.BadSpec: The query does not contain model `EntityType`
```

File: src/dispatch/static/dispatch/src/signal/SignalInstanceTab.vue [hunk 3]
```diff
@@ -9,12 +9,6 @@
     <template v-slot:item.signal="{ item }">
       <signal-popover v-model="item.signal" />
     </template>
-    <template v-slot:item.duplication_rule="{ item }">
-      <v-simple-checkbox :value="item.duplication_rule || false" disabled></v-simple-checkbox>
-    </template>
-    <template v-slot:item.suppression_rule="{ item }">
-      <v-simple-checkbox :value="item.supression_rule || false" disabled></v-simple-checkbox>
-    </template>
     <template v-slot:item.created_at="{ item }">
```
Comment (wssheldon) [inline src/dispatch/static/dispatch/src/signal/SignalInstanceTab.vue:12]:
```suggestion
    <template v-slot:item.entities="{ item }">
      <v-row>
        <v-chip v-for="(entity, index) in item.entities" :key="index" class="mr-2">
          {{ entity.entity_type.name }}
        </v-chip>
      </v-row>
    </template>
    <template v-slot:item.created_at="{ item }">
```

Maybe something like this, shows `object Object` right now

Comment (wssheldon) [reply src/dispatch/static/dispatch/src/signal/SignalInstanceTab.vue:12]: Something like this is probably better if we have a lot of them:

```vue
<template v-slot:item.entities="{ item }">
  <template v-for="(entity, index) in item.entities.slice(0, 3)">
    <v-chip :key="index" label>{{ entity.entity_type.name }}</v-chip>
  </template>
  <v-chip v-if="displayEntityCount(item.entities)" label>+{{ item.entities.length - 3 }}</v-chip>
</template>

<script>
export default {
  // ...
  methods: {
    displayEntityCount(entities) {
      return entities.length > 3;
    },
  },
};
</script>
```

File: src/dispatch/signal/service.py (hunk 2 above)
Comment (kevgliss) [reply src/dispatch/signal/service.py:231]: This should be covered by `query = apply_filter_specific_joins(SignalInstance, f.expression, query)` I'd be interested to know when you encountered this error?

Comment (wssheldon) [reply src/dispatch/signal/service.py:231]: The error was raised when I tested signal ingestion end-to-end w/ a filter applied, using the scheduled consumer task.

**Pull request merged**